src/egon/data/importing/openstreetmap/import_osm.py

```diff
@@ -0,0 +1,219 @@
+
+    # Prepare osm2pgsql command
+    cmd = [
+        "osm2pgsql",
+        "--create",
+        "--slim",
+        "--hstore-all",
+        f"--number-processes {num_processes}",
+        f"--cache {cache_size}",
+        f"-H {docker_db_config['HOST']} -P {docker_db_config['PORT']} "
+        f"-d {docker_db_config['POSTGRES_DB']} "
+        f"-U {docker_db_config['POSTGRES_USER']}",
+        f"-p {osm_config['table_prefix']}",
+        f"-S {osm_config['stylefile']}",
+        f"{osm_config['file']}",
+    ]
+
+    # Execute osm2pgsql for import OSM data
+    subprocess.run(
+        " ".join(cmd),
+        shell=True,
+        env={"PGPASSWORD": docker_db_config["POSTGRES_PASSWORD"]},
+        cwd=os.path.dirname(__file__),
+    )
+
+
+def post_import_modifications():
+    """Adjust primary keys, indices and schema of OSM tables.
+
+    * The Column "gid" is added and used as the new primary key.
+    * Indices (GIST, GIN) are reset
+    * The tables are moved to the schema configured as the "output_schema".
+    """
+    # Replace indices and primary keys
+    for table in [
+        "osm_" + suffix for suffix in ["line", "point", "polygon", "roads"]
+    ]:
+
+        # Drop indices
+        sql_statements = [f"DROP INDEX {table}_index;"]
+
+        # Drop primary keys
+        sql_statements.append(f"DROP INDEX {table}_pkey;")
+
+        # Add primary key on newly created column "gid"
+        sql_statements.append(f"ALTER TABLE public.{table} ADD gid SERIAL;")
+        sql_statements.append(
+            f"ALTER TABLE public.{table} ADD PRIMARY KEY (gid);"
+        )
+        sql_statements.append(
+            f"ALTER TABLE public.{table} RENAME COLUMN way TO geom;"
+        )
+
+        # Add indices (GIST and GIN)
+        sql_statements.append(
+            f"CREATE INDEX {table}_geom_idx ON public.{table} "
+            f"USING gist (geom);"
+        )
+        sql_statements.append(
+            f"CREATE INDEX {table}_tags_idx ON public.{table} "
+            f"USING GIN (tags);"
+        )
+
+        # Execute collected SQL statements
+        for statement in sql_statements:
+            db.execute_sql(statement)
+
+    # Get dataset config
+    data_config = egon.data.config.datasets()["openstreetmap"][
+        "original_data"
+    ]["osm"]
+
+    # Move table to schema "openstreetmap"
+    db.execute_sql(
+        f"CREATE SCHEMA IF NOT EXISTS {data_config['output_schema']};"
+    )
+
+    for out_table in data_config["output_tables"]:
+        sql_statement = (
+            f"ALTER TABLE public.{out_table} "
+            f"SET SCHEMA {data_config['output_schema']};"
+        )
+
+        db.execute_sql(sql_statement)
+
+
+def metadata():
+    """Writes metadata JSON string into table comment."""
+    # Prepare variables
+    osm_config = egon.data.config.datasets()["openstreetmap"]["original_data"][
+        "osm"
+    ]
+    spatial_and_date = os.path.basename(osm_config["file"]).split("-")
```

Comment:
Note to self: marked as resolved because we're using `osm_config["original_data"]["target"]["path"]` now, which can be a path (duh) which makes `os.path.basename` important again.